helm/idol-library/templates/aciserver/_service.tpl

```diff
@@ -0,0 +1,40 @@
+# BEGIN COPYRIGHT NOTICE
+# Copyright 2023 Open Text.
+# 
+# The only warranties for products and services of Open Text and its affiliates and licensors
+# ("Open Text") are as may be set forth in the express warranty statements accompanying such
+# products and services. Nothing herein should be construed as constituting an additional warranty.
+# Open Text shall not be liable for technical or editorial errors or omissions contained herein.
+# The information contained herein is subject to change without notice.
+#
+# END COPYRIGHT NOTICE
+{{- define "idol-library.aciserver.service.base" -}}
+{{/*
+    Common template for a generic ACI server service
+*/}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Values.name | quote }}
+  labels: {{- include "idol-library.labels" . | nindent 4 }}
+spec:
+  ports:
+  - port: {{ .Values.aciPort | int }}
+    targetPort: aci-port
+    name: aci-port
```

Comment:
Worth including an optional index port block here? (Guarded with `{{- if .Values.indexPort }}` or similar.)

Reply:
Guess that'd need a corresponding block in the deployment template.

Reply:
probably only want an index port for statefulsets?

Reply:
though addiing to the service definition would be fine

Reply:
> probably only want an index port for statefulsets?

Fair point.

> though addiing to the service definition would be fine

App selector in the service would work with either a deployment or a stateful set, right?